### PR TITLE
crypto-utils: Replace TextEncoder dependency

### PR DESCRIPTION
**Pull request by someone1**

### CODE REVIEW

**Comment by someone1** [comment]
Performance bump too (not that this is used much in the library): https://jsperf.com/str2ab-vs-textencoder

File: src/crypto_utils.ts [hunk 1]
```diff
@@ -35,6 +34,17 @@ export function urlSafe(buffer: Uint8Array): string {
   return encoded.replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '');
 }
 
+// adapted from source: http://stackoverflow.com/a/11058858
+export function str2ab(str: string) {
```
Comment (tikurahul) [inline src/crypto_utils.ts]: Lets call this method something like `textEncodeLite()`. 
Also you should add a comment that describes why we did this in the first place with a caniuse link.

Comment (someone1) [reply src/crypto_utils.ts]: Will do - thanks!

**Comment by someone1** [comment]
I've signed the CLA, thanks for the feedback and please let me know if there's anything else!

**Comment by someone1** [comment]
FYI - You have one failing test related to the test code "challengechallengechallengechallenge" - its fails as follows: "FAILED Crypto Utils Tests. produces the right base64 encoded challenge" - the code needs to be between 43 and 128 characters. This is unrelated to the changes I introduced.

Does it make sense to add some sort of CI to this repository so it can auto-run tests on every PR?

**Comment by tikurahul** [comment]
Thanks for letting me know about the test. I knew about it, but have been swamped with other things.

The PR looks good. :+1:

**Comment by uifox** [comment]
Hey @tikurahul,
Can you please let me know when this change will be published to the npm repo? We were also facing the unavailable TextEncoder problem and I ended up pointing to the HEAD version of AppAuth-JS in the github repo such that we can get this change, however we would prefer pointing to an explicit version for more predictable and reliable dependency management. Thanks.

**Pull request merged**